tests/test_parse.py

```diff
@@ -30,3 +35,95 @@ def test_float(self):
     def test_bool(self):
         c = parse(Config, ["--a", "12", "--d", "True"])
         assert c.d is True
+
+    def test_list(self):
+        c = parse(Config, ["--a", "12", "--e", "1", "2", "3"])
+        assert c.e == [1, 2, 3]
+
+
+class TestParseLists:
+    def test_parse_list_default(self):
+        @dataclass
+        class TConfig:
+            arg: list[str] = field(default_factory=lambda: [])
+
+        t = parse(TConfig, [])
+        assert t.arg == []
+
+    def test_parse_list(self):
+        @dataclass
+        class TConfig:
+            arg: list[str] = field(default_factory=lambda: [])
+
+        t = parse(TConfig, ["--arg", "1", "2"])
+        assert t.arg == ["1", "2"]
+
+    def test_parse_list_int(self):
+        @dataclass
+        class TConfig:
+            arg: list[int] = field(default_factory=lambda: [])
+
+        t = parse(TConfig, ["--arg", "1", "2"])
+        assert t.arg == [1, 2]
+
+    def test_parse_list_required(self):
+        @dataclass
+        class TConfig:
+            arg: list[int] = field()
```

Comment:
is this allowed? isn't default_factory required for lists when they are non-default?

Reply:
This field has no default.